### PR TITLE
Redesign UI and introduce dark theme

### PR DESCRIPTION
**Pull request by msfjarvis**
- Refactor all hardcoded color references to use theme attributes
- Switch color names to be more descriptive of their purpose
- There is no user-facing toggle for this at present, but it will
  follow the system dark theme on Q and battery saver on P and below.

### CODE REVIEW

**Comment by MSF-Jarvis** [comment]
The app UI is a dumpster fire so this PR will also land a full UI redesign.

**Comment by MSF-Jarvis** [comment]
screenshot-20190607-133815
screenshot-20190607-133716

File: .travis.yml [hunk 1]
```diff
@@ -13,6 +13,10 @@ android:
   licenses:
     - ".+"
 
+before_install:
+  - mkdir -p "$ANDROID_HOME/licenses/"
+  - curl "https://download.msfjarvis.website/.viscerion/sdk-licenses.tar.xz" | tar -C "$ANDROID_HOME/licenses/" --strip-components=1 -xvf -
```
Comment (zidhuss) [inline .travis.yml]: Is there a way to do this without downloading from your personal website?

Comment (MSF-Jarvis) [reply .travis.yml]: Yes, I'll be changing that, just copied a quick-fix from an internal repo to resolve CI failure.

Comment (MSF-Jarvis) [reply .travis.yml]: Done with 0971a72

**Comment by zeapo** [comment]
Fuck, you're the best!

**Comment by MSF-Jarvis** [comment]
> Fuck, you're the best!

I try :)

Still a lot of work left to do but we're in a good shape at the moment

**Comment by MSF-Jarvis** [comment]
Pending changes

- Figure out how to re-add the broken preference from OpenKeychain's integration library
- String improvements across the board to better document features and explain input fields
- ???

**Comment by 532910** [comment]
@msfjarvis, please take into consideration #444

**Comment by msfjarvis** [comment]
@zidhuss @zeapo this is still blocked by OpenKeychain's library not updating to AndroidX and thus breaking every modern caller. Any insights on what should be our path forward?

**Pull request merged**